packages/eslint-plugin/tests/rules/no-unnecessary-template-expression.test.ts

```diff
@@ -162,6 +162,48 @@ ruleTester.run('no-unnecessary-template-expression', rule, {
         },
       ],
     },
+    {
+      code: '`${0o25}`;',
+      output: '`21`;',
+      errors: [
+        {
+          messageId: 'noUnnecessaryTemplateExpression',
+          line: 1,
+          column: 4,
+          endColumn: 8,
+        },
+      ],
+    },
+    {
+      code: '`${0b1010} ${0b1111}`;',
+      output: '`10 15`;',
```

Comment:
part of me wonders if we should not handle these weird number representation cases and just leave them as is.

eg I could see more value in writing code with a bunch of variations on the binary string to ensure completeness vs having the converted int val.

for example
```ts
`${0b00} ${0b01} ${0b10} ${0b11}` 
```
encodes information better (i.e. that all binary cases are represented) rather than
```ts
`0 1 2 3`
```
Obviously this is a simplified case but one could imagine some complex binary sequence represented (eg power of two or something).
Same reason that it's not uncommon for people to define flag enums as
```
enum Flag {
  A = 1 << 0,
  B = 1 << 1,
  C = 1 << 2,
  D = 1 << 3,
  E = 1 << 4,
}

// rather than

enum Flag {
  A = 1,
  B = 2,
  C = 4,
  D = 8,
  E = 16,
}
```

probably just put a pin in it and leave it as a potential future exercise

Reply:
Interesting point, yea, I can imagine someone would want their binary/hex numbers to remain untouched. But this PR already has a lot of changes beyond the original issue. So I'd say we should file a follow-up issue or wait for someone to report it.

> probably just put a pin in it and leave it as a potential future exercise

:+1: +1 here